src/input/README.md

```diff
@@ -0,0 +1,188 @@
+# Input Component
+
+## Usage
+
+### Basic usage
+
+```javascript
+import * as React from 'react';
+import {StatefulInput} from 'baseui/input';
+
+export default () => {
+  return
+    <StatefulInput
+      value={props.name}
+      onChange={evt => props.setName(evt.target.value)}
+      onKeyDown={evt => {
+        if (evt.key === 'Enter') {
+          props.setName('');
+        }
+      }}
+    />;
+};
+```
+
+### Advanced usage
+
+```javascript
+import {StatefulInput as Input, StyledRoot} from 'baseui/input';
+import {withStyle} from 'styletron-react';
+
+const RootWithStyle = withStyle(StyledRoot, props => {
+  const {
+    $isFocused,
+    $theme: {sizing},
+  } = props;
+  return {
+    borderRadius: $isFocused ? '0' : sizing.scale100,
+  };
+});
+
+export default () => {
+  return (
+    <div>
+      <Input overrides={{Root: RootWithStyle}} />
+    </div>
+  );
+};
+```
+
+## Exports
+
+* `StatefulInput`
+* `StatefulContainer`
+* `Input`
+* `BaseInput`
+* `StyledRoot`
+* `StyledInputEnhancer`
+* `StyledInputContainer`
+* `StyledInput`
+* `ADJOINED`
+* `CUSTOM_INPUT_TYPE`
+* `SIZE`
+* `STATE_CHANGE_TYPE`
+
+## `StatefulInput` API
+
+* All properties of the StatefulContainer except `children`
+* All properties of the Input except `$components` that are optional and used to pass overrides for default styled components set.
+
+## `StatefulContainer` API
+
+* `children: (props: Props) => React.Node`
+* `initialState: {value: string} = {value: ''}`
+  * Initial state of an uncontrolled input component.
+    * `value` - an initial input value.
+* `stateReducer: (type: $Keys<STATE_CHANGE_TYPE>', nextState: {}, currentState: {}, e: SyntheticEvent<HTMLInputElement>) => stateToSet: {}` - Optional
```

Comment:
Ideally the `$Keys<STATE_CHANGE_TYPE>` should be  changed to`$Values`. But similar comment here about using `STATE_CHANGE_TYPE.change | null` instead of  `$Keys<STATE_CHANGE_TYPE>`.